src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_inpaint_legacy.py

```diff
@@ -509,6 +509,11 @@ def __call__(
 
         if not isinstance(mask_image, torch.FloatTensor):
             mask_image = preprocess_mask(mask_image, self.vae_scale_factor)
+        else:
+            # permute dimensions to pytorch standard (B, C, H, W)
+            mask_image = mask_image.permute(0, 3, 1, 2)
```

Comment:
Thanks! Could we add a test here?

Reply:
I think we should do some additional checks here. What if `maks_image` is already `channels_first` ?

Reply:
Good point, user may pass mask channel first in standard shape. I'm thinking of the test below for expected mask input mask shape (B x 1 x H x W) or (B x H x W x 1). 
Also worth considering, do you want to support user passing mask tensor with 3 channel dimensions? (B x 3 x H x W)?

`import torch

image = torch.randn(1,3,512,512)
mask = torch.randn(1,512,512,1)

# squeeze potential singleton channel dimensions, then unsqueeze to pytorch standard (B x 1 x H x W)
mask = mask.squeeze(3).squeeze(1).unsqueeze(1)

if mask.shape[-2:] != image.shape[-2:]:
    raise ValueError(
            f"Mask height and width {tuple(mask.shape[-2:])} should match image height and width {tuple(image.shape[-2:])}"
        )
mask.shape`